kubernetes-ingress/tls-with-cert-manager.md

```diff
@@ -0,0 +1,121 @@
+# Ingress example with TLS termination using cert-manager
+
+This example builds on the [HTTP](http.md) and [gRPC](grpc.md) ingress
+examples, adding TLS termination. This example is similar to the
+[TLS](tls.md) ingress example, in which you had to manually create a
+self-signed CA using `minica`. In this example, you will be using
+[cert-manager](https://cert-manager.io/), a Kubernetes-native operator that
+issues and renews TLS certificates for you.
+
+## Create TLS certificate and private key
+
+Let us install cert-manager:
+
+```sh
+helm repo add jetstack https://charts.jetstack.io
+helm install cert-manager jetstack/cert-manager --version v1.7.1 --namespace cert-manager --set installCRDs=true --create-namespace
+```
+
+Now, create a CA Issuer:
+
+```sh
+kubectl apply -f https://raw.githubusercontent.com/cilium/cilium-service-mesh-beta/main/kubernetes-ingress/ca-issuer.yaml
+```
+
+## Deploy the ingress
+
+If you previously ran the HTTP and/or gRPC ingress demos, delete the ingresses:
+
+```sh
+kubectl delete ingress basic-ingress grpc-ingress
+```
+
+The ingress configuration for this demo provides the same routing as those demos
+but with the addition of TLS termination. Deploy this ingress:
+
+```sh
+kubectl apply -f https://raw.githubusercontent.com/cilium/cilium-service-mesh-beta/main/kubernetes-ingress/tls-ingress.yaml
+```
```

Comment:
This is using the same ingress config as the "static certificate" example, which makes me wonder whether it matters that there is a `tls.secretName` configured in tls-ingress.yaml? Could/should the user delete that `secretName` field from the yaml and it would work? 

If the user had previously been through the "static" example would they need to delete that secret? I wonder if you can end up thinking you're using a cert-manager cert, but actually it's using the previous one that happened to be lying around?

Reply:
Excellent point.

After having completed the tls.md guide, one would end up with the secret `demo-cert` left over.

And since `demo-cert` already exists, one may wonder: will cert-manager still issue a new certificate and overwrite the existing `demo-cert`. That's actually what cert-manager does: when a Secret already exists, it updates it in-place when the newly issued certificate.

I will:
- re-run the tutorial locally and see whether cert-manager properly re-issues the certificate when the secret already exists,
- document the fact that the existing demo-cert, if the tutorial tls.md was run previously, will be updated with cert-manager's issued certificate.

Let me know if I should rather use another secret name, e.g. `cert-demo-2`.